scripts/globals/helm.lua

```diff
@@ -1402,6 +1402,9 @@ tpz.helm.onTrade = function(player, npc, trade, helmType, csid)
     local zoneId = player:getZoneID()
     local regionId = player:getCurrentRegion()
 
+    -- HELM should remove invisible
+    player:delStatusEffect(tpz.effect.INVISIBLE)
```

Comment:
Can someone confirm that HELM actions drop invis?  IIRC, the player must drop invis manually before being able to mine for example.